client/client.go

```diff
@@ -266,50 +277,49 @@ func (c *HostClient) syncRevision(ctx context.Context, contractID types.FileCont
 // reports an invalid signature, suggesting the local revision is out of sync,
 // it will synchronize with the host and retry the function using the updated
 // revision. Therefore, the revise function must be idempotent.
-func (c *HostClient) withRevision(ctx context.Context, contractID types.FileContractID, reviseFn func(revision types.V2FileContract) (types.V2FileContract, error)) error {
+func (c *HostClient) withRevision(ctx context.Context, contractID types.FileContractID, reviseFn func(revision types.V2FileContract) (rhp.ContractRevision, error)) error {
```

Comment:
```suggestion
func (c *HostClient) withRevision(ctx context.Context, contractID types.FileContractID, reviseFn func(revision rhp.ContractRevision) (rhp.ContractRevision, error)) error {
```

This is better. That way the input and output of the passed in function are the same so the caller doesn't need to construct it. Especially since RPCs expect a `ContractRevision` anyway.